code/controllers/subsystem/tcgsetup.dm

```diff
@@ -0,0 +1,10 @@
+#define CARD_FILES list("set_one.json","set_two.json")
+#define CARD_DIRECTORY "strings/tcg"
+#define CARD_PACKS list(/obj/item/cardpack/series_one, /obj/item/cardpack/resin)
+SUBSYSTEM_DEF(trading_card_game)
+	name = "Trading Card Game"
+	flags = SS_NO_FIRE
+
+/datum/controller/subsystem/trading_card_game/Initialize()
+	reloadAllCardFiles(CARD_FILES, CARD_DIRECTORY)
+	. = ..()
```

Comment:
Add a proc called on the parent and redefined on the child instead.

Reply:
What do you mean? Isn't this what Initialize is for?

Reply:
Yes, but you are adding behavior to happen before initialization.
In general you want to run the parent first on init, and later when you have a guarantee things are somewhat setup the child's code.

Reply:
This is what the subsystem parent looks like 
![image](https://user-images.githubusercontent.com/58055496/82845939-67fd6280-9e9b-11ea-8db7-4b16c9921dfa.png)
And looking at airss this seems to be the intended use.
![image](https://user-images.githubusercontent.com/58055496/82845957-7fd4e680-9e9b-11ea-8b68-d14df4722af1.png)
I'm not totally familiar with the subsystem well system. It seems to use initialize as a "We have finished loading", so I think this is a valid use

Reply:
Fair, then `return ..()` at the end.

Reply:
gottcha